### DISCUSSION TITLE
nativescript ios on Mac M1 ld: framework not found FirebaseCore clang: error: linker command failed with exit code 1 (use -v to see invocation)

### DISCUSSION THREAD

**Discussion by GuineaPing** [NativeScript/NativeScript#10378] [Q&A] [answered]
### Issue Description

It happen after migration live nativescript project from Mac Intel to new Mac M1.
When I building for iOS on step: Link project (arm64)
getting the next: 
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/AMPopTip'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/CocoaLumberjack'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/DAScratchPad'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/FirebaseCore'

etc...
for all pods
then

ld: framework not found FirebaseCore
clang: error: linker command failed with exit code 1 (use -v to see invocation)

### Reproduction

1. Move nativescript project from Mac Intel to Mac M1 Pro
2. Install the same versions for node, npm & tns
3. tns run ios
4. It seems any nativescript project fails to buid on new M1 Pro

### Relevant log output (if applicable)

```shell
....
Merged 1100 categories.
Result: 55894 declarations from 177 top level modules
Done! Running time: 5.54415 sec 
Saving metadata generation's stderr stream to: /Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/metadata-generation-stderr-arm64.txt
~/Projects/XCode/Haultail/native-script/haultail-new-0/haultail/platforms/ios
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/AMPopTip'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/CocoaLumberjack'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/DAScratchPad'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/FirebaseCore'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/FirebaseCoreDiagnostics'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/FirebaseInstallations'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/FirebaseInstanceID'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/FirebaseMessaging'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/GoogleDataTransport'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/GoogleUtilities'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/IQKeyboardManager'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/ISMessages'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/MBProgressHUD'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/MDFInternationalization'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/MMNumberKeyboard'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/MaterialComponents'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/OMGHTTPURLRQ'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/PocketSocket'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/PromisesObjC'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/Protobuf'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/QBImagePickerController'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/SAMKeychain'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/SVGKit'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/Sentry'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/SocketRocket'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/SwiftyCam'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/TrustKit'
ld: warning: directory not found for option '-F/Users/eugenelysenko/Library/Developer/Xcode/DerivedData/haultail-gwqydtrltqtzrgdgfjrvopcijwcv/Build/Products/Debug-iphonesimulator/nanopb'
ld: framework not found FirebaseCore
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### Environment

OS: macOS 13.4.1
CPU: (10) x64 Apple M1 Pro
Shell: /bin/zsh
node: 14.17.6
npm: 8.19.3
nativescript: 8.5.3

# android
java: 11.0.13
ndk: Not Found
apis: 28, 29, 30, 31, 32, 33
build_tools: 29.0.2, 30.0.2, 30.0.3, 31.0.0
system_images: 
  - android-29 | Google APIs ARM 64 v8a
  - android-30 | ARM 64 v8a
  - android-30 | Google APIs ARM 64 v8a
  - android-30 | Google Play ARM 64 v8a
  - android-31 | ARM 64 v8a
  - android-31 | Google APIs ARM 64 v8a
  - android-31 | Google Play ARM 64 v8a
  - android-32 | Google APIs ARM 64 v8a
  - android-32 | Google Play ARM 64 v8a
  - android-33 | Google APIs ARM 64 v8a
  - android-33 | Google Play ARM 64 v8a
  - android-TiramisuPrivacySandbox | Google Play ARM 64 v8a
  - android-UpsideDownCake | Google APIs ARM 64 v8a

# ios
xcode: 14.3.1/14E300c
cocoapods: 1.12.1
python: 2.7.18
python3: 3.9.6
ruby: 2.7.5
platforms: 
  - DriverKit 22.4
  - iOS 16.4
  - macOS 13.3
  - tvOS 16.4
  - watchOS 9.4

"dependencies": {
	"@angular/animations": "~8.2.0",
	"@angular/cli": "^9.0.6",
	"@angular/common": "~8.2.0",
	"@angular/compiler": "~8.2.0",
	"@angular/core": "~8.2.0",
	"@angular/forms": "~8.2.0",
	"@angular/platform-browser": "~8.2.0",
	"@angular/platform-browser-dynamic": "~8.2.0",
	"@angular/router": "~8.2.0",
	"@exeleon/nativescript-svg": "^1.0.1",
	"@klippa/nativescript-http": "^1.0.0",
	"@mapbox/polyline": "^1.1.1",
	"@nativescript-community/gps": "^2.1.1",
	"@nativescript/geolocation": "^7.1.1",
	"@nativescript/theme": "~2.3.3",
	"@nstudio/nativescript-camera-plus": "^2.2.7",
	"@nstudio/nativescript-loading-indicator": "^3.0.4",
	"creditcards-types": "^2.2.1",
	"decode-google-map-polyline": "^1.0.1",
	"email-validator": "^2.0.4",
	"geojson-mend": "^0.1.1",
	"html-to-text": "^5.1.1",
	"moment-strftime": "^0.5.0",
	"moment-timezone": "^0.5.26",
	"nativescript-android-sensors": "^1.5.0",
	"nativescript-angular": "^8.21.0",
	"nativescript-app-environment": "^1.0.0",
	"nativescript-appversion": "^1.4.4",
	"nativescript-barcodescanner": "^3.4.2",
	"nativescript-camera": "^4.5.0",
	"nativescript-dna-deviceinfo": "^2.4.2",
	"nativescript-feedback": "^1.3.10",
	"nativescript-fingerprint-auth": "^7.0.2",
	"nativescript-fonticon": "^2.0.0",
	"nativescript-google-maps-sdk": "^2.7.0",
	"nativescript-imagepicker": "^7.1.0",
	"nativescript-insomnia": "^1.2.3",
	"nativescript-iqkeyboardmanager": "^1.5.1",
	"nativescript-loading-indicator-new": "^2.1.0",
	"nativescript-masked-text-field": "^4.0.3",
	"nativescript-md5": "^1.0.1",
	"nativescript-ngx-shadow": "^6.5.0",
	"nativescript-ngx-slides": "^6.1.0",
	"nativescript-numeric-keyboard": "^4.3.1",
	"nativescript-paint": "^1.0.0",
	"nativescript-permissions": "^1.3.9",
	"nativescript-perms": "^2.0.11",
	"nativescript-phone": "^1.4.1",
	"nativescript-plugin-firebase": "^10.5.2",
	"nativescript-secure-storage": "^2.5.0",
	"nativescript-sensors": "^1.0.0",
	"nativescript-sentry": "^1.10.3",
	"nativescript-theme-core": "~1.0.6",
	"nativescript-tooltip": "^2.1.2",
	"nativescript-ui-autocomplete": "^6.0.0",
	"nativescript-videoplayer": "^4.2.1",
	"nativescript-websockets": "^1.5.5",
	"nativescript-windowed-modal": "^6.2.1",
	"node-sass": "^4.13.1",
	"parse-url": "^5.0.1",
	"password-validator": "^5.0.2",
	"reflect-metadata": "~0.1.12",
	"rxjs": "^6.4.0",
	"rxjs-compat": "^6.5.3",
	"tns-core-modules": "^6.5.2",
	"tns-platform-declarations": "^6.5.1",
	"zone.js": "~0.9.1"
	},
"devDependencies": {
	"@angular/compiler-cli": "~8.2.0",
	"@ngtools/webpack": "8.2.0",
	"nativescript-dev-webpack": "^1.5.1",
	"tns-android": "6.5.3",
	"tns-ios": "6.5.6",
	"typescript": "~3.5.3"
	},

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

**Comment by rigor789** [comment, marked as answer]
Seems like this is an old project, pre Apple Silicon era. Some of the plugins used don't have arm64 architecture slices in them to run natively on the new mac/simulators.

One option is to run on physical devices - which should work because the device slices haven't changed with the introduction of Apple Silicon. You might need to replace `tns-ios: 6.5.6` with `@nativescript/ios@8.x` (latest) which should work fine regardless of the rest of your project dependencies.

Alternatively, you need to either update the project to latest, including replacing deprecated plugins like nativescript-plugin-firebase, or use something like `patch-package` to patch incompatible plugins directly after identifying them.

For example, for firebase specifically, something like [this](https://github.com/EddyVerbruggen/nativescript-plugin-firebase/issues/1867#issuecomment-1188780572) might work. (this is where `patch-package` comes in handy, because you can persist these changes to `node_modules` inside patch files, and re-apply them after reinstalling `node_modules`).

_PS: If you need help with a project upgrade, you can always reach out to one of the professional partners: https://nativescript.org/partners/ for a quote_
